mr/mapper_executor.cc

```diff
@@ -194,15 +192,15 @@ void MapperExecutor::IOReadFiber(detail::TableBase* tb) {
                file_record_cnt = uint64_t{0}](string&& s) mutable {
       if (file_record_cnt++ < skip)
         return;
-      record_q.Push(Record::RECORD, aux_local->records_read, std::move(s));
-      ++aux_local->records_read;
+      record_q.Push(Record::RECORD, aux_local->raw_context->metric_map()["fn-calls"], std::move(s));
```

Comment:
i have a bug in the original logic here.
you should pass here: `file_record_cnt - skip` imho instead of `aux_local->raw_context->metric_map()["fn-calls"]`.
The original intent was to pass the record position inside the current stream...
I use this attribute, for example, in mrgrep.